### PR TITLE
CORE-1363 Add pending termination dlg

### PR DESCRIPTION
**Pull request by ash3rz**
This displays when users try to go to an output folder (from either a button or menu item in the Analyses view) that doesn't yet exist because the analysis is still running.

image

### CODE REVIEW

File: src/components/analyses/PendingTerminationDlg.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import React from "react";
+
+import { build } from "@cyverse-de/ui-lib";
+import { Button, Typography } from "@material-ui/core";
+
+import DEDialog from "components/utils/DEDialog";
+import { useTranslation } from "i18n";
+import ids from "./ids";
+
+function PendingTerminationDlg(props) {
```
Comment (ash3rz) [inline src/components/analyses/PendingTerminationDlg.js:17]: Not sure if "Pending Termination" makes sense to anybody else - open to suggestions on the name

File: src/components/analyses/listing/Actions.js [hunk 2]
```diff
@@ -49,16 +50,29 @@ const RelaunchButton = React.forwardRef((props, ref) => {
...
-            href={href}
-            onClick={onClick}
-            ref={ref}
+            href={isTerminated ? href : null}
+            ref={isTerminated ? ref : null}
+            onClick={(event) => {
+                if (isTerminated) {
+                    onClick();
+                } else {
+                    setPendingTerminationDlgOpen(true);
```
Comment (ash3rz) [inline src/components/analyses/listing/Actions.js:73]: I would have used `stopPropagation` and `preventDefault` here, but then the row doesn't get selected, which means I don't have access to the analysis name. So I opted for disabling the `href` and `ref` instead

File: src/components/analyses/utils.js [hunk 3]
```diff
@@ -210,6 +210,22 @@ const canShare = (selectedAnalyses) => {
     );
 };
 
+const ANALYSIS_TERMINAL_STATES = [
+    analysisStatus.CANCELED,
+    analysisStatus.COMPLETED,
+    analysisStatus.FAILED,
+    analysisStatus.REMOVED,
```
Comment (ash3rz) [inline src/components/analyses/utils.js:217]: I'm not 100% sure if these are all the terminal states from [here](https://github.com/cyverse-de/sonora/blob/master/src/components/models/analysisStatus.js)

Comment (sri2k1us) [reply src/components/analyses/utils.js:217]: LGTM 👍🏽

**Review by sri2k1us — APPROVED** [review]
I have 1 suggestion about the message displayed to the user. But LGTM 👍🏽

**Review by psarando — APPROVED** [review]
LGTM :shipit: 👍

**Comment by ash3rz** [comment]
I just realized I forgot to include this change for the dashboard action as well.  I'll make that change before merging.

**Comment by ash3rz** [comment]
Thanks for the reviews!

**Pull request merged**